torch/serialization.py

```diff
@@ -1112,6 +1116,22 @@ def load_tensor(dtype, numel, key, location):
         name = f'data/{key}'
 
         storage = zip_file.get_storage_from_record(name, numel, torch.UntypedStorage)._typed_storage()._untyped_storage
+        # check if byteswapping is needed and swap here if it is needed
+        byteencodingname = 'byteencoding'
+        if zip_file.has_record(byteencodingname):
```

Comment:
This is getting checked for every storage loaded from a file, we should probably only do it once per file if possible

Reply:
It looks like loading uses python unpickler. How can I pass this data through it?

Reply:
I think you can just move the `zip_file.get_record()` call above the `def load_tensor(...):` line, something like this:

```python
byteorderdata = None
if zip_file.has_record(byteordername):
            byteorderdata = zip_file.get_record(byteordername)
```

Reply:
Thanks, will try that out.